api_end_end_testing/src/test/java/tds/student/api/tests/StudentApiTests.java

```diff
@@ -0,0 +1,726 @@
+    }
+
+    @Test
+    public void shouldNotCreateUpdateStudentWithInvalidMiddleName() {
+        String ssid = createRandomStudentSsid();
+        String stateAbbrev = "CA";
+
+        // Create a student
+        StudentInfo studentInfo = createStudent(ssid, stateAbbrev);
+
+        // Test an invalid middle name
+        studentInfo.setMiddleName("MymiddlenameisWilliam-Peterson-Henderson");
+
+        given()
+            .contentType(ContentType.JSON)
+            .header(authHeader)
+            .body(studentInfo)
+        .when()
+            .post(uriLocation + stateAbbrev)
+        .then()
+            .statusCode(400)
+            .body("messages.middleName[0]", equalTo("The max size of the MiddleName is 35"));
+    }
+
+    @Test
+    public void shouldNotCreateBatchOfStudentsWithInvalidBirthDate() {
+        String ssid1  = createRandomStudentSsid();
+        String ssid2  = createRandomStudentSsid();
+        String stateAbbrev = "CA";
+        List<StudentInfo> students = createBatchOfStudents(ssid1, ssid2, stateAbbrev);
+
+        // Test an invalid birth date
+        students.get(0).setBirthDate("882292-12-983");
+
+        given()
+            .contentType(ContentType.JSON)
+            .header(authHeader)
+            .body(students)
+        .when()
+            .post(uriLocation + stateAbbrev + "/" + "batch")
+        .then()
+            .statusCode(400);
+    }
+
+    @Test
+    public void shouldNotCreateUpdateStudentWithInvalidExternalSsid() {
+        String ssid = createRandomStudentSsid();
+        String stateAbbrev = "CA";
+
+        // Create a student
+        StudentInfo studentInfo = createStudent(ssid, stateAbbrev);
+
+        // Test and invalid state abbreviation
+        studentInfo.setExternalSsid("");
+
+        given()
+            .contentType(ContentType.JSON)
+            .header(authHeader)
+            .body(studentInfo)
+        .when()
+            .post(uriLocation + stateAbbrev)
+        .then()
+            .statusCode(400);
+    }
+
+    @Test
+    public void shouldNotCreateUpdateStudentWithInvalidSex() {
+        String ssid = createRandomStudentSsid();
+        String stateAbbrev = "CA";
+
+        // Create a student
+        StudentInfo studentInfo = createStudent(ssid, stateAbbrev);
+
+        // Test and invalid state abbreviation
+        studentInfo.setSex("");
+
+        given()
+            .contentType(ContentType.JSON)
+            .header(authHeader)
+            .body(studentInfo)
+        .when()
+            .post(uriLocation + stateAbbrev)
+        .then()
+            .statusCode(400);
+    }
+
+    @Test
+    public void shouldNotCreateUpdateStudentWithInvalidGradeLevel() {
+        String ssid = createRandomStudentSsid();
+        String stateAbbrev = "CA";
+
+        // Create a student
+        StudentInfo studentInfo = createStudent(ssid, stateAbbrev);
+
+        // Test and invalid state abbreviation
```

Comment:
Same comment about comment not matching what is being done.  All of them need to be looked at and updated to reflect what is being done.